app/interactors/access_limit/update_interactor.rb

```diff
@@ -28,14 +28,18 @@ def find_and_lock_edition
   def update_access_limit
     limit_type = params.require(:limit_type)
 
-    if LIMIT_TYPES.include?(limit_type)
-      context.fail! if edition.access_limit&.limit_type == limit_type
-      edition.assign_access_limit(limit_type, user)
+    if LIMIT_TYPES.exclude?(limit_type)
+      context.fail! if edition.access_limit.nil?
+      edition.assign_as_edit(user, access_limit: nil)
       return
     end
 
-    context.fail! if edition.access_limit.nil?
-    edition.remove_access_limit(user)
+    access_limit = AccessLimit.build_initial(created_by: user,
+                                             edition: edition,
+                                             limit_type: limit_type)
+
+    context.fail! if edition.access_limit&.limit_type == limit_type
```

Comment:
Feels like this guard clause would be more intuitive before the object creation.

Reply:
Sure - moved it 👍 .